### ISSUE TITLE
Add accessibility support for color-blindness

### ISSUE THREAD

**Issue by tobythomassen** [chakra-ui/chakra-ui#2701]
# Feature request

## Is your feature request related to a problem? Please describe.

This is not a code problem, but still a valid concern. The issue can be broken down into two parts: 1) color variants such as "success" and "warning" should change their native colors and or provide even more visual feedback as to the state 2) theme colors should be adjusted in hue for different types of color-blindness.

## Describe the solution you'd like

Adding a new hook is a sound solution similar to how useColorMode() works for changing to dark theme. The proposed hook would take a variable representing the type of color blindness found [here](https://www.nei.nih.gov/learn-about-eye-health/eye-conditions-and-diseases/color-blindness/types-color-blindness), and then it would modify the color pallet accordingly.

## Describe alternatives you've considered

Currently, the best way to handle this has just been to add custom variants, but this can be very tedious, and it is only one more reason for developers to skip it entirely.

## Additional context

There is not much support for people with color-blindness, and it is not very well standardized across the web. Most developers either don't consider it as an issue or handle it in entirely different ways.

Here are some sources from w3 that explain the issue and some possible solutions a bit:
https://www.w3.org/TR/UNDERSTANDING-WCAG20/visual-audio-contrast-without-color.html
https://www.w3.org/TR/low-vision-needs/

**Comment by tobythomassen** [comment]
I don't really have the time to follow up on this; however, I do think it is something that needs to be addressed at some point. I really hope that someone is able to work on it in the near future.
